sklearn/metrics/_plot/tests/test_plot_confusion_matrix.py

```diff
@@ -247,20 +246,47 @@ def test_confusion_matrix_pipeline(pyplot, clf, data, n_classes):
     assert disp.text_.shape == (n_classes, n_classes)
 
 
-@pytest.mark.parametrize("values_format", ['e', 'n'])
+@pytest.mark.parametrize("values_format", [None, 'e', 'd', 'n'])
 def test_confusion_matrix_text_format(pyplot, data, y_pred, n_classes,
                                       fitted_clf, values_format):
+
+    def test_function_format(x):
+        if (x == 0 or np.log10(x) < 7):
+            return 'd'
+        else:
+            return '.2g'
+
     # Make sure plot text is formatted with 'values_format'.
     X, y = data
+
     cm = confusion_matrix(y, y_pred)
     disp = plot_confusion_matrix(fitted_clf, X, y,
                                  include_values=True,
                                  values_format=values_format)
 
     assert disp.text_.shape == (n_classes, n_classes)
 
-    expected_text = np.array([format(v, values_format)
-                              for v in cm.ravel()])
-    text_text = np.array([
-        t.get_text() for t in disp.text_.ravel()])
-    assert_array_equal(expected_text, text_text)
+    if values_format is None:
```

Comment:
Having a big if/else statement like this kinda defeats the purpose of parametrizing the test.

Maybe keep this test unchanged and create a new one `test_confusion_matrix_text_format_default` which would only check the default behavior.